Description: Fix cast warnings
 This patch fixes some casting warnings that come up in the
 Debian build process.
Author: Roland Stigge <stigge@antcom.de>

--- dia-0.97.2.orig/app/sheets_dialog_callbacks.c
+++ dia-0.97.2/app/sheets_dialog_callbacks.c
@@ -1159,7 +1159,7 @@ sheets_dialog_togglebutton_set_sensitive
   if (is_sensitive)
   {
     tmp = lookup_widget(dialog, "button_ok");
-    gtk_object_set_data(GTK_OBJECT(tmp), "active_type", (gpointer)type);
+    gtk_object_set_data(GTK_OBJECT(tmp), "active_type", (gpointer)(intptr_t)type);
   }
 
   for (i = 0; widget_names[i]; i++)
--- dia-0.97.2.orig/app/layer_dialog.c
+++ dia-0.97.2/app/layer_dialog.c
@@ -1449,7 +1449,7 @@ undo_layer_visibility(Diagram *dia, Laye
 
   for (i = 0; i < layers->len; i++) {
     Layer *temp_layer = (Layer *) g_ptr_array_index(layers, i);
-    visibilities = g_list_append(visibilities, (gpointer)temp_layer->visible);
+    visibilities = g_list_append(visibilities, (gpointer)(intptr_t)temp_layer->visible);
   }
 
   change->original_visibility = visibilities;
--- dia-0.97.2.orig/app/menus.c
+++ dia-0.97.2/app/menus.c
@@ -1165,7 +1165,7 @@ menus_set_recent (GtkActionGroup *action
 
   do {
     id = gtk_ui_manager_new_merge_id (ui_manager);
-    recent_merge_ids = g_slist_prepend (recent_merge_ids, (gpointer) id);
+    recent_merge_ids = g_slist_prepend (recent_merge_ids, (gpointer)(intptr_t) id);
 
     gtk_ui_manager_add_ui (ui_manager, id, 
                  recent_path, 
